source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,107 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+--------------------------
+Speculative Authentication
+--------------------------
+
+:since: 4.4
+
+The ``isMaster`` handshake supports a new argument, ``speculativeAuthenticate``,
+provided as a BSON object. Clients specifying this argument to ``isMaster`` will
+speculatively include the first command of an authentication handshake.
+This command may be provided to the server in parallel with any standard request for
+supported authentication mechanisms (i.e. ``saslSupportedMechs``). This would permit
+clients to merge their first message with their ``isMaster`` request, and
+receive the authentication reply with the ``isMaster`` reply.
+
+When the mechanism is ``MONGODB-X509``, `speculativeAuthenticate` has the following
+structure::
+
+    {
+        isMaster: 1,
+        speculativeAuthenticate: {
+            authenticate: 1,
+            mechanism: "MONGODB-X509"
+        }
+    }
+
+When the mechanism is ``SCRAM-SHA-1`` or ``SCRAM-SHA-256``, `speculativeAuthenticate`
+has the following structure::
+
+    {
+        isMaster: 1,
+        speculativeAuthenticate: {
+            saslStart: 1,
+            mechanism: "<string>",   /* "SCRAM-SHA-1" or "SCRAM-SHA-256" */
+            db: "<authentication database>",
+            payload: BinData(...),
+            options: {
+                skipEmptyExchange: true         /* REQUIRED */
+            }
+        }
+    }  
+
+
+speculativeAuthenticate.mechanism
```

Comment:
As above, I think all this can go away.

Reply:
I have removed the descriptions of the fields.